idea-plugin/src/main/java/com/palantir/javaformat/intellij/PalantirJavaFormatFormattingService.java

```diff
@@ -89,10 +89,17 @@ public void run() {
             }
 
             try {
-                String formattedText = applyReplacements(
-                        request.getDocumentText(),
-                        formatterService.get().getFormatReplacements(request.getDocumentText(), toRanges(request)));
-                request.onTextReady(formattedText);
+                List<Replacement> replacements =
+                        formatterService.get().getFormatReplacements(request.getDocumentText(), toRanges(request));
+
+                // The Javadoc of onTextReady API says that you should set it to null when the
+                // document is unchanged. But an even better version is to simply not attempt
+                // to format a document that is already formatted
+                if (replacements.isEmpty()) {
```

Comment:
Damn so it would turn out that even when the file is fully formatted it still produces a replacement which looks like:

![image](https://github.com/user-attachments/assets/8c673392-48fa-44b3-adff-b552db425420)

A full file replacement with the same content... Seems like a bug in the formatter service.

Reply:
To avoid having to remote debug your intellij, here is a failing test that exemplifies the issue: https://github.com/palantir/palantir-java-format/pull/1188/files#diff-a6622bd2788dface780176f19bbaecce6308bb0a826e6e6721b0c713bc0e2c68R450